### DISCUSSION TITLE
HealthScope CRD is missing

### DISCUSSION THREAD

**Discussion by dhiguero** [kubevela/kubevela#1696] [Q&A]
**Describe the bug**

On a fresh installation of Kubevela, the `HealthScope` CRD has not been registered.

**To Reproduce**

1. Follow the [Getting Started](https://kubevela.io/docs/install) to install Kubevela on Kind.
2. List scopes

```
$ kubectl get scopedefinitions.core.oam.dev
No resources found in default namespace.
$ kubectl get healthscopes.core.oam.dev
No resources found in default namespace.
```

**Expected behavior**

Since the runtime implements an operator for the _HealthScope_, the CRD should be generated and registered in the system during the installation.

**Screenshots**

**Cluster information**

* Kind cluster created following the getting started with `kindest/node:v1.18.15`.

**Additional context**

After applying the health scope definition from the [oam-kubernetes-runtime](https://github.com/crossplane/oam-kubernetes-runtime) I was able to obtain the scope as expected

<details>
<summary>$ kubectl get healthscopes.core.oam.dev app-health -oyaml</summary>

```
apiVersion: core.oam.dev/v1alpha2
kind: HealthScope
metadata:
  creationTimestamp: "2021-04-14T16:54:04Z"
  generation: 2
  managedFields:
  - apiVersion: core.oam.dev/v1alpha2
    fieldsType: FieldsV1
    fieldsV1:
      f:spec: {}
    manager: kubectl-create
    operation: Update
    time: "2021-04-14T16:54:04Z"
  - apiVersion: core.oam.dev/v1alpha2
    fieldsType: FieldsV1
    fieldsV1:
      f:spec:
        f:workloadRefs: {}
      f:status:
        .: {}
        f:healthConditions: {}
        f:scopeHealthCondition:
          .: {}
          f:healthStatus: {}
          f:healthyWorkloads: {}
          f:total: {}
    manager: manager
    operation: Update
    time: "2021-04-14T16:54:09Z"
  name: app-health
  namespace: default
  resourceVersion: "5968"
  selfLink: /apis/core.oam.dev/v1alpha2/namespaces/default/healthscopes/app-health
  uid: 5b6966e0-ae45-4527-b9db-43560377f3e2
spec:
  workloadRefs:
  - apiVersion: apps/v1
    kind: Deployment
    name: express-server
status:
  healthConditions:
  - componentName: express-server
    diagnosis: 'Ready:1/1 '
    healthStatus: HEALTHY
    targetWorkload:
      apiVersion: apps/v1
      kind: Deployment
      name: express-server
      uid: 1f732f24-6c2d-4355-b813-55c9c14076a5
  scopeHealthCondition:
    healthStatus: HEALTHY
    healthyWorkloads: 1
    total: 1
```

</details>

**Comment by zzxwill** [comment]
@dhiguero Thanks for reporting this issue.

The healthscope in KubeVela is in namespaced scope https://github.com/oam-dev/kubevela/blob/master/charts/vela-core/templates/definitions/healthscopes.yaml#L5, while it is in cluster scope in OAM Kubernetes Runtime https://github.com/crossplane/oam-kubernetes-runtime/blob/master/charts/oam-kubernetes-runtime/templates/scopedefinitions.yaml.

So you may check healthscope by `kubectl get healthscopes.core.oam.dev -n vela-system`. As you installed KubeVela by default, `vela-system` should be the system namespace.

And could you please show me your application manifest, I can run it myself. Thanks.

**Comment by dhiguero** [comment]
Thanks for the clarification, I have tried again from a fresh install, and this is the output I get:

```
$ kubectl -n default get scopedefinitions.core.oam.dev
No resources found in default namespace.
$ kubectl -n test get scopedefinitions.core.oam.dev
No resources found in test namespace.
$ kubectl -n vela-system get scopedefinitions.core.oam.dev
NAME                        DEFINITION-NAME
healthscopes.core.oam.dev   healthscopes.core.oam.dev
```

It is true that the definition is installed in vela-system, but I would expect `healthscopes.core.oam.dev` to be "available" as `scopedefinitions.core.oam.dev` in every namespace as with the previous oam-kubernetes-runtime given that it is a default core scope.

Could you please elaborate on this change from the previous behavior? 

By the way, this is the application I was using for the test (based on the existing example from the getting started):
<details>
<summary>app-with-scope.yaml</summary>

```apiVersion: core.oam.dev/v1alpha2
kind: HealthScope
metadata:
  name: app-health
spec:
  workloadRefs: []
---
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: first-vela-app
spec:
  components:
    - name: express-server
      type: webservice
      properties:
        image: crccheck/hello-world
        port: 8000
      scopes:
        "healthscopes.core.oam.dev": "app-health"

```
</details>

**Comment by wonderflow** [comment]
@dhiguero Hi, the definitions in  `vela-system` can work in any namespace. This change was introduced in https://github.com/oam-dev/kubevela/pull/1085

And the issue is https://github.com/oam-dev/kubevela/issues/918

In the design, `vela-system` is the KubeVela system level namespace, any definitions there can be widely used in any namespaces as system built-in capability. While other namespaces are user level namespace, definitions there can be seperated by user. 

\cc @zzxwill
